### PR TITLE
Data class rule: removing several cases from the scope of the inspection

### PR DESCRIPTION
**Pull request by akuleshov7**
### What's done:
- Removed annotation, value, sealed, inline, inner classes from the scope
- Added tests

### CODE REVIEW

**Review by sanyavertolet — APPROVED** [review]

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/DataClassesRule.kt [hunk 1]
```diff
@@ -128,6 +127,12 @@ class DataClassesRule(configRules: List<RulesConfig>) : DiktatRule(
         return true
     }
 
+    /** we do not exclude inner classes and enums here as if they have no
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/DataClassesRule.kt:130]: I think we filter out inner classes and enums elsewhere:
https://github.com/saveourtool/diktat/blob/70f46ca6509d20ecf524b199bf8845809325a51f/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/DataClassesRule.kt#L99-L100

**Review by petertrr — APPROVED** [review]

**Review by nulls — APPROVED** [review]

**Review by 0x6675636b796f75676974687562 — APPROVED** [review]

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/DataClassesRule.kt (hunk 1 above)
Comment (akuleshov7) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/DataClassesRule.kt:130]: fuck

Comment (akuleshov7) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/DataClassesRule.kt:130]: also I have no idea, why "open" is prohibited

Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/DataClassesRule.kt:130]: image

Comment (akuleshov7) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/DataClassesRule.kt:130]: but what is the idea? You have open class only with data and properties?

Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/DataClassesRule.kt]: If there are no abstract fields, than diktat will suggest to change `abstract` to `open` ¯\\_(ツ)_/¯

**Pull request merged**